File: 1399/ex.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println(countLargestGroup(24))
}

func countLargestGroup(n int) int {
	result := 0
	var curMap [][]int = make([][]int, n)

	for i := range curMap {
		curMap[i] = make([]int, n)
	}

	for i := 1; i <= n; i++ {
		num := i
		localSum := 0

		for num > 9 {
			localSum += num % 10
			num = (num - num%10) / 10
		}

		localSum += num

		curMap[localSum] = append(curMap[localSum], i)
	}

	highest := 0

	for _, element := range curMap {
		if len(element) > highest {
			highest = len(element)
			result = 0
		}

		if len(element) == highest {
			result += 1
		}
	}

	return result
}

File: 1399/ex.ts
function countLargestGroup(n: number): number {
    let result = 0;
    let map: number[][] = [];

    for (let i = 1; i <= n; i++) {
        let num = i;
        let localSum = 0;

        while (num > 9) {
            localSum += num % 10;
            num = (num - num % 10) / 10;
        }
        localSum += num;

        map[localSum] ? map[localSum].push(i) : map[localSum] = [i];
    }

    let highest = 0;
    map.forEach(m => {
        if (m.length > highest) {
            highest = m.length;
            result = 0;
        }

        m.length == highest ? result++ : null;
    });

    console.debug(map);

    return result;
};

// console.log(countLargestGroup(13));
// console.log(countLargestGroup(2));
console.log(countLargestGroup(24)); // 5